R/zchunk_L145.water.demand.municipal.R

```diff
@@ -43,18 +47,72 @@ module_water_L145.water.demand.municipal_DISABLED <- function(command, ...) {
     # ===================================================
     # TRANSLATED PROCESSING CODE GOES HERE...
     #
-    # If you find a mistake/thing to update in the old code and
-    # fixing it will change the output data, causing the tests to fail,
-    # (i) open an issue on GitHub, (ii) consult with colleagues, and
-    # then (iii) code a fix:
-    #
-    # if(OLD_DATA_SYSTEM_BEHAVIOR) {
-    #   ... code that replicates old, incorrect behavior
-    # } else {
-    #   ... new code with a fix
-    # }
-    #
-    #
+    # Aggregate FAO_municipal_water_AQUASTAT to GCAM regions and fill out the years for missing data.
+
+    #Map FAO country to GCAM region ID
+    iso_GCAM_regID %>% select(iso, GCAM_region_ID) %>%
+      inner_join(AGLU_ctry[,c("iso","FAO_country")], by ="iso") -> FAO_ctry_GCAM_region_ID
+
+    FAO_municipal_water_AQUASTAT %>%
+      rename ( FAO_country =Area ) %>%
+      inner_join(FAO_ctry_GCAM_region_ID[, c( "GCAM_region_ID", "FAO_country" ) ], by ="FAO_country") -> ctry_municipal_W
+
+    #Aggregate to regions and fill out missing years using rule=2
+    ctry_municipal_W %>%
+      group_by(GCAM_region_ID, Year) %>%
+      summarise(Value = sum(Value)) %>%
+      ungroup() %>%
+      complete(GCAM_region_ID = unique(iso_GCAM_regID$GCAM_region_ID),
+               Year = HISTORICAL_YEARS,
+               fill = list(Value = NA))%>%
+      group_by(GCAM_region_ID)%>%
+      mutate(Value = approx_fun(Year, Value, rule=2)) %>%
+      rename(value = Value, year =Year) %>%
+      mutate(water_type ="water withdrawals")-> municipal_water_R_W_Yh_km3
+
+    # Come up with GCAM reginal average prices starting with the country level IBNET data.
+    # Note since the years are all over the place we will just use the average across years too.
+    #Come up with regional municipal prices from IBNET data
+    IBNET_municipal_water_cost_USDm3 %>%
+      mutate(FAO_country = country)%>%
+      inner_join(FAO_ctry_GCAM_region_ID, by="FAO_country")%>%
+      mutate(expenditure = cost * consumption)  %>%
+      group_by(GCAM_region_ID)%>%
+      summarise(expenditure = sum(expenditure), consumption=sum(consumption))%>%
+      mutate(input.cost = expenditure/consumption)%>%
+      select(GCAM_region_ID, input.cost)-> municipal_water_cost_R_75USD_m3
+
+    # The IBNET data is incomplete and so it is very possible that we have entire GCAM regions in which none
+    # of the contained countries had any reported data.
+    stopifnot( nrow( municipal_water_cost_R_75USD_m3 ) == nrow( GCAM_region_names ) )
+
+    # Map water use efficiencies from the continent scale to GCAM regions
+    # Note names were changed to match manufacturing continent to avoid the need for an additional mapping
+
+    #create the variable year2 for temporal interpolation
+    year2 <- sort(c(1900, 1940, 1950, 1960, 1970, 1980, 1990, 1995, 2000,modeltime.YEARS[-2]))
```

Comment:
Very confusing what's going on here. What are these years? Why are you dropping the second element of `modeltime.YEARS`?